### PR TITLE
ENH: Add GOLD Nmax data support

### PR DESCRIPTION
**Pull request by jklenzing**
Adding basic support for the Nmax data product from GOLD.  Provides nighttime plasma density and irradiance data.  Cleaning not yet implemented, so a warning is in place.

### CODE REVIEW

**Comment by jklenzing** [comment]
When you load the data into xarray, it shows up with three dimensions: nscan, nlat, nlon.  Each scan takes place over a finite period of time (see https://svs.gsfc.nasa.gov/4610).  Currently, I'm providing the timestamp for the start of each scan as the "epoch" for the gold data.  There might be a more intuitive way of doing this, but if not this should be documented in the routines.

File: pysatNASA/instruments/gold_nmax.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
+""" Supports the Nmax data product from the GOLD spacecraft
+
```
Comment (jklenzing) [inline pysatNASA/instruments/gold_nmax.py:2]: add full documentation relative to other insts

**Comment by jklenzing** [comment]
Look into separating north and south images, see if that solves the strict_time_flag problem

**Comment by jklenzing** [comment]
Requires pysat/pysat#598 to merge

**Comment by jklenzing** [comment]
Ready for review.  Failure related to travis.org migration.  Will add test code shortly.

**Comment by jklenzing** [comment]
```
import pysat
from pysatNASA.instruments import gold_nmax
import matplotlib.pyplot as plt

plt.ion()
gold = pysat.Instrument(inst_module=gold_nmax, strict_time_flag=False)
gold.load(2020, 1)
scan = 0
plt.pcolor(gold['longitude'][scan, :, :], gold['latitude'][scan, :, :], gold['nmax'][scan, :, :])
plt.xlabel('Longitude')
plt.ylabel('Latitude')
plt.title('Sample GOLD Nmax data')
```
Figure_1

Incrementing the scan number allows you to tab through the various images.

**Comment by jklenzing** [comment]
Regarding the strict_time_flag, I'm still not the right way to proceed.  Options:
- reshape data structure to split northern and southern images.  
  - timestamp refers to start of image scan.  Since images do not necessarily start at the same time, this decouples physical timestamps from one of the hemispheres
- Drop half the data, using a tag to specify north or south
  - Problematic for scanning through large data sets. Would have to load data twice during automated search for plasma structure
- Leave as is

Poke through the data and let me know if you think of other routes forward.  `strict_time_flag=False` should get me through near-term projects, but may not be a long-term solution.

**Comment by aburrell** [comment]
I am for the first option, but adding 'northern' and 'southern' time variables so that no info is lost.

**Comment by aburrell** [comment]
After updating your example to download data, I was able to reproduce the figure.  The base Instrument representation looks like:

```
pysat Instrument object
-----------------------
Platform: 'gold'
Name: 'nmax'
Tag: ''
Instrument id: ''

Data Processing
---------------
Cleaning Level: 'clean'
Data Padding: None
Keyword Arguments Passed to list_remote_files: {'remote_url': 'https://cdaweb.gsfc.nasa.gov', 'user': None, 'password': None, 'fake_daily_files_from_monthly': False, 'two_digit_year_break': None, 'delimiter': None}
Custom Functions: 0 applied

Local File Statistics
---------------------
Number of files: 2
Date Range: 01 January 2020 --- 02 January 2020

Loaded Data Statistics
----------------------
Date: 01 January 2020
DOY: 001
Time range: 01 January 2020 20:10:48 --- 02 January 2020 00:25:23
Number of Times: 24
Number of variables: 23

Variable Names:
mask_oi_1356     mask_wavelength  radiance_oi_1356 
                        ...                        
oi_1356_unc_sys  hemisphere       time             

pysat Meta object
-----------------
Tracking 23 metadata values
Metadata for 22 standard variables
Metadata for 0 ND variables
```

I have to say, I am unsure about the naming convention used here.  After looking here: https://hpde.io/NASA/NumericalData/GOLD/L2/NMAX/PT900S.html I expected the name to be 'NI1'.  Why did you end up settling on 'nmax' instead?

**Comment by jklenzing** [comment]
'NI1' is the Level 1 data set from which this data is constructed.  There's a separate archive for that on CDAweb.

We should probably have a tag up over data standards (coords, etc) with both of us looking at the file.

**Comment by jklenzing** [comment]
Reverting to draft until standards are finalized

**Comment by jklenzing** [comment]
Standards are up to date (thanks @aburrell!).  Moving forward with warnings in place for non-unique / non-monotonic timestamps.  Full solution requires updates to pysat, documented in pysat/pysat#614.

File: pysatNASA/instruments/__init__.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 __all__ = ['cnofs_ivm', 'cnofs_plp', 'cnofs_vefi',
            'de2_lang', 'de2_nacs', 'de2_rpa', 'de2_wats',
-           'formosat1_ivm',
+           'gold_nmax', 'formosat1_ivm',
```
Comment (aburrell) [inline pysatNASA/instruments/__init__.py:3]: This should be 'ses1_gold', right?

**Review by aburrell — COMMENTED** [review]
Partial review since I think the platform, name, and tag should be updated.

File: pysatNASA/instruments/__init__.py (hunk 2 above)
Comment (jklenzing) [reply pysatNASA/instruments/__init__.py]: updated

File: pysatNASA/instruments/__init__.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
            'de2_lang', 'de2_nacs', 'de2_rpa', 'de2_wats',
            'formosat1_ivm',
            'icon_euv', 'icon_fuv', 'icon_ivm', 'icon_mighti',
-           'iss_fpmu', 'omni_hro', 'sport_ivm',
+           'iss_fpmu', 'omni_hro', 'ses1_gold', 'sport_ivm',
```
Comment (aburrell) [inline pysatNASA/instruments/__init__.py:5]:
```suggestion
           'iss_fpmu', 'omni_hro', 'ses14_gold', 'sport_ivm',
```

I misread the NASA page.  Here's the sc info: https://space.skyrocket.de/doc_sdat/ses-14.htm

File: pysatNASA/instruments/ses1_gold.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
+"""Supports the Nmax data product from the Global Observations of the Limb and
+Disk (GOLD) satellite.  Accesses data in netCDF format.
+
+Properties
+----------
+platform
+    'ses1'
```
Comment (aburrell) [inline pysatNASA/instruments/ses1_gold.py:7]:
```suggestion
    'ses14'
```

File: pysatNASA/instruments/ses1_gold.py [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+- strict_time_flag must be set to False
+
+
+Examples
+--------
+::
+
+    import datetime as dt
+    import pysat
+    nmax = pysat.Instrument(platform='ses1', name='gold', tag='nmax'
```
Comment (aburrell) [inline pysatNASA/instruments/ses1_gold.py:25]:
```suggestion
    nmax = pysat.Instrument(platform='ses14', name='gold', tag='nmax'
```

File: pysatNASA/instruments/ses1_gold.py [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+from pysat import logger
+from pysat.instruments.methods import general as ps_gen
+from pysat.utils import load_netcdf4
+from pysatNASA.instruments.methods import gold as mm_gold
+from pysatNASA.instruments.methods import cdaweb as cdw
+
+# ----------------------------------------------------------------------------
+# Instrument attributes
+
+platform = 'ses1'
```
Comment (aburrell) [inline pysatNASA/instruments/ses1_gold.py:50]:
```suggestion
platform = 'ses14'
```

File: pysatNASA/instruments/ses1_gold.py [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+name = 'gold'
+tags = {'nmax': 'Nmax data for the GOLD instrument'}
```
Comment (aburrell) [inline pysatNASA/instruments/ses1_gold.py:52]:
```suggestion
tags = {'nmax': 'Level 2 Nmax data for the GOLD instrument'}
```

File: pysatNASA/instruments/ses1_gold.py [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+    logger.info(mm_gold.ack_str)
+    logger.warning(' '.join(('Time stamps may be non-unique because Channel A',
+                             'and B are different instruments.  An upgrade to',
+                             'the pysat.Constellation object is required to',
+                             'solve this issue. See pysat issue #614 for more',
+                             'info.')))
+    self.acknowledgements = mm_gold.ack_str
+    self.references = mm_gold.ref_str
+
+    pass
```
Comment (aburrell) [inline pysatNASA/instruments/ses1_gold.py:87]:
```suggestion
    return
```

Because this is the end of the method.

**Review by aburrell — APPROVED** [review]
Yay!  I look forward to the final version of this instrument, but this will do for now.

**Pull request merged**